### PR TITLE
Eventonica React Lesson Pt 1

### PR DESCRIPTION
**Pull request by dpenny**
Revamp the Eventonica Project to use React
In this lesson, participants will:

Create a Users component with:
- A list using a mock list of user objects
- A simple form to create a user with an ID, name and email (and other fields if they'd like)
- A DeleteUser component that will delete a user from the list

This will be followed up by part 2 (outline included). Part 2 will be a more advanced react lesson, and will introduce useReducer, useEffect, and possibly some other react concepts, depending on timing

- the starter code is adjusted a bit to format the inline instructions as comments, and move the `user` section before the `events` section, since participants will work on the User flow first. 

Screenshot 2021-08-29 at 21 27 23

### CODE REVIEW

File: projects/eventonica-react/eventonica-react-ui.md [hunk 1]
```diff
@@ -42,101 +57,132 @@ It's typical for a React developer to start in one file, then begin to divide it
...
+  setUsers([...users, newUser]);
+};
 ```
+**JS Syntax Checks**: 
+Take a look at some of the object and array syntax in the code snippet above. Do you understand what `[...users, newUser]` represents?
 
-### Display All Users
+`const newUser = {id: id, name: name, email: email}` can also be written as
+`const newUser = {id, name, email}`
+This is a ES2015 feature called [Object property shorthand](https://alligator.io/js/object-property-shorthand-es6/)
```
Comment (alodahl) [inline projects/eventonica-react/eventonica-react-ui.md:127]: 💕

File: projects/eventonica-react/eventonica-react-part-2.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- Add a variable to save which user is currently "logged in", so saving an event for a user doesn't require typing in the user's ID every time.
+- Use localStorage to store your data so it doesn't get deleted every time the page is refreshed. Learn about localStorage from the [MDN localStorage docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)
+- Change the UI. Draw up a design for how you'd like your app to look and then code your site to match. The changes could involve breaking the UI into multiple pages, adding more CSS, using a styling library such as Bootstrap, and/or adding more JS. Look at websites that have successful UIs for creating, retrieving, updating, deleting, and filtering data for inspiration. Consider designing a desktop and a mobile version and create a responsive UI.
+- Deploy your site so you can share it with others. Deploying means putting your site on the internet where other people can access it. One easy way to deploy this site is to use GitHub pages. Follow this tutorial to deploy your site: https://www.codecademy.com/articles/f1-u3-github-pages.
```
Comment (alodahl) [inline projects/eventonica-react/eventonica-react-part-2.md:40]: they have done this before, so deploying a client-side site could be an option here. however, we will be adding the backend within a week of this outline and soon deploying it with heroku (unlike in the past when it was several weeks later), so this step is not essential.

**Review by alodahl — APPROVED** [review]
i love part 2!  maybe useReducer and next steps can be the mvp, and you can add the rest when you have time?

**Comment by alodahl** [comment]
Question:  why are there two PRs for this?  I am lost help

**Comment by dpenny** [comment]
oops closed the old one, my b!
I'll merge this now then open a new PR for useReducer?

**Pull request merged**